quadsim/robots/motors.py

```diff
@@ -1,48 +1,54 @@
-""" Implements models for single DC-DC motors and compositional motor groups.
-"""
+"""Implements models for single DC motors and compositional motor groups."""
 import enum
+from dataclasses import dataclass
 from typing import Optional
 from typing import Sequence
 from typing import Tuple
 from typing import Union
 
 import numpy as np
 
-POSITION_INDEX = 0
-KP_INDEX = 1
-VELOCITY_INDEX = 2
-KD_INDEX = 3
-TORQUE_INDEX = 4
-COMMAND_DIMENSION = 5
+_ARRAY = Sequence[float]
+_FloatOrArray = Union[float, _ARRAY]
```

Comment:
Is the purpose of this to make the code more readable?

Reply:
Readability and avoid duplicate code.

Reply:
Makes sense!

I'm not sure I like the 'code smell' of these type aliases, but I do agree we should address this since it's bound to come up all over the place.

Check this out as a potential option: https://github.com/ramonhagenaars/nptyping

There might be other better solutions too.